contracts/src/assertionStakingPool/AssertionStakingPool.sol

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+// SPDX-License-Identifier: BUSL-1.1
+//
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../rollup/IRollupLogic.sol";
+import "../rollup/IRollupCore.sol";
+import "./StakingPoolErrors.sol";
+
+/// @notice Creates staking pool contract for a target assertion. Can be used for any child Arbitrum chain running on top of the deployed AssertionStakingPoolCreator's chain.
+contract AssertionStakingPoolCreator {
+    event NewAssertionPoolCreated(
+        address indexed rollup,
+        bytes32 indexed _assertionHash,
+        address assertionPool
+    );
+
+    /// @notice Create a staking pool contract
+    /// @param _rollup Rollup contract of target chain
+    /// @param _assertionInputs Inputs to be passed into Rollup.stakeOnNewAssertion
+    /// @param _assertionHash Assertion hash to be passed into Rollup.stakeOnNewAssertion
+    function createPoolForAssertion(
+        address _rollup,
+        AssertionInputs memory _assertionInputs,
+        bytes32 _assertionHash
+    ) external returns (address) {
+        address assertionPoolAddress = address(
+            new AssertionStakingPool(_rollup, _assertionInputs, _assertionHash)
+        );
```

Comment:
let's use CREATE2 here to prevent duplicated staking pool, we can also have a getter that return the staking pool if one already existed